pkg/job_controller/job.go

```diff
@@ -324,6 +330,108 @@ func (jc *JobController) ReconcileJobs(job interface{}, replicas map[apiv1.Repli
 	return result, nil
 }
 
+func (jc *JobController) CreateCronJob(meta metav1.Object, runtimeObj runtime.Object, policy *apiv1.RunPolicy) error {
+	cronJob := &appv1.Cron{}
+	isCronJob := policy != nil && policy.CronPolicy != nil
+	if err := jc.Client.Get(context.Background(), types.NamespacedName{
+		Namespace: meta.GetNamespace(),
+		Name:      meta.GetName(),
+	}, cronJob); err != nil {
+		if errors.IsNotFound(err) {
+			if !isCronJob {
+				//Do not need to run regularly
+				return nil
+			}
+			instance, err := jc.createCronInstance(policy, runtimeObj)
+			if err != nil {
+				return err
+			}
+			return jc.Client.Create(context.Background(), instance)
+		}
+		return err
+	}
+	if !isCronJob {
+		//Job no longer needs to run regularly, delete the originally created cronJob
+		return jc.Client.Delete(context.Background(), cronJob)
+	}
+	//If cronPolicy or cronTemplate change, the cronJob needs to be updated
+	newCronJob := cronJob.DeepCopy()
+	needUpdate := false
+	if !reflect.DeepEqual(cronJob.Spec.CronPolicy, *policy.CronPolicy) {
+		newCronJob.Spec.CronPolicy = *policy.CronPolicy
+		needUpdate = true
+	}
+	newInstance, err := jc.createCronInstance(policy, runtimeObj)
+	if err != nil {
+		return err
+	}
+	if !reflect.DeepEqual(cronJob.Spec.CronTemplate.Workload.Raw, newInstance.Spec.CronTemplate.Workload.Raw) {
+		newCronJob.Spec.CronTemplate = newInstance.Spec.CronTemplate
+		needUpdate = true
+	}
+	if needUpdate {
+		return jc.Client.Patch(context.Background(), newCronJob, client.MergeFrom(cronJob))
+	}
+	return nil
+}
+
+func (jc *JobController) createCronInstance(policy *apiv1.RunPolicy, runtimeObj runtime.Object) (*appv1.Cron, error) {
+	tempPolicy := policy.DeepCopy()
+	defer func() {
+		policy.CronPolicy = tempPolicy.CronPolicy
+	}()
+	// Clear the CronPolicy under the cronjob created by simulation
+	// If not cleared, the cronJob created here will continue to
+	// create a cronJob, resulting in a dead cycle
+	policy.CronPolicy = nil
+	kind := runtimeObj.GetObjectKind().GroupVersionKind().Kind
+	apiversion := runtimeObj.GetObjectKind().GroupVersionKind().GroupVersion().String()
+	extCodec := pkgruntime.NewRawExtensionCodec(jc.Scheme)
+	newMeta := runtimeObj.DeepCopyObject().(metav1.Object)
+	metaName := newMeta.GetName()
+	metaNs := newMeta.GetNamespace()
+	// The new Cron only need Spec info, but Spec cannot get here,
+	// it can only be converted into metaObj by force, then cleared
+	// some attributes of meta. The overall framework will be refactored
+	// and optimized in the future
+	clearMetaInfoForCronTemplate(newMeta)
+	rawObj, err := extCodec.EncodeRaw(newMeta.(runtime.Object))
```

Comment:
extCodec.EncodeRaw(runtimeObject)

Reply:
Need to encode a cleaned-up runtimeObj, the input runtimeObj contains some parameters that should not be inherited